src/segment.rs

```diff
@@ -387,17 +386,16 @@ impl Segment {
     }
 
     /// Flushes recently written entries to durable storage.
-    pub fn flush_async(&mut self) -> Future<(), Error> {
+    pub fn flush_async(&mut self, runtime: Handle) -> Option<tokio::task::JoinHandle<Result<()>>> {
```

Comment:
Why not do it like this and spawn at the call site?
```suggestion
    pub async fn flush_async(&mut self, runtime: Handle) -> Result<()> {
```

Also are there any blocking operations here? In that case I would proceed as you have done, but [spawn_blocking](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#method.spawn_blocking) instead